### PR TITLE
LPS-145927

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
@marianoalvarosaiz no regressions found.

File: portal-kernel/test/unit/com/liferay/portal/kernel/util/LocaleUtilTest.java [hunk 1]
```diff
@@ -130,6 +130,25 @@ public void testFromLanguageIdConsistency() {
...
+
+		languageUtil.setLanguage(language);
+
+		Mockito.when(
+			language.isAvailableLocale(Locale.ITALY)
+		).thenReturn(
+			false
+		);
+
+		Assert.assertSame(
```
Comment (tinatian) [inline portal-kernel/test/unit/com/liferay/portal/kernel/util/LocaleUtilTest.java:166]: @marianoalvarosaiz You commit message says "Ensure that a locale is returned if no validation is performed", but you test code did not assert a locale is returned. You only asserted you get same result when you invoked it twice, but you won't know if it is null or not.

Comment (tinatian) [reply portal-kernel/test/unit/com/liferay/portal/kernel/util/LocaleUtilTest.java:166]: @marianoalvarosaiz Forget it, I fixed it in the later commits.

Tina

**Comment by tinatian** [comment]
@marianoalvarosaiz Code looks good to me, please rebase and resend. This pull is big, could you please try to separate it into small pulls? This way you may not have that many code conflicts.

Thanks.
Tina

**Comment by marianoalvarosaiz** [comment]
@tinatian see https://github.com/liferay-core-infra/liferay-portal/pull/1225. Only last four commits sent first because are the ones that generate the semantic versioning change that is susceptible to conflicts. Afterwards I'll continue sending the others.

**Pull request closed without merging**